src/OrbitBase/include/OrbitBase/StopState.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_STOP_STATE_H_
+#define ORBIT_BASE_STOP_STATE_H_
+
+#include <absl/synchronization/mutex.h>
+
+namespace orbit_base_internal {
+
+// StopState is an implementation detail of StopSource and StopToken and is not intended to be used
+// outside of StopSource and StopToken
+class StopState {
```

Comment:
It seems to me like this entire class could be replaced by an "std::shared_ptr<atomic<bool>> shared_stop_requested_", in StopToken.

Reply:
Yes, it could be, specifically `std::atomic<bool>`. I thought and read up about it when you brought it up in the last review. I should've addressed it, but forgot about it, my mistake. I think in this specific instance an atomic would be actually fine and not a problem, but I decided against it anyways. go/atomic-danger discourages the use of atomic, but mainly talks about people trying to build lock free data structures and how easy it is to get things wrong. This is not really applicable here, because this would be a fairly simple use of std::atomic. Then there is go/atomic-safety, with the section "The 'simple atomic value' pattern", which is what could be done in `StopToken` and `StopSource`. Thats why I think it would be okay to use an atomic here. But the page is also in general rather opposed to use atomic instead of mutexes, for example: "Mutex protected values should not be changed to std::atomic values without data confirming the cost saving opportunity." 
I am usually a fan of making things smaller, but in this instance I think it is valuable to have the extra class. A reader might wonder why an atomic is used and if there are any implications. The whole code of StopState can be read in just a moment and everyone easily understands that this is just a simple value made thread safe with a mutex. (And the small detail that a StopState can only be set to `true` once). So I am sticking with the StopState class